File: Java/pattern/Momento/src/com/pattern/Originator.java
package com.pattern;

public class Originator {
	private String status;
	
	public Momento createMomento() {
		return new Momento(status);
	}
	
	public void restoreStatus(Momento momento) {
		this.status = momento.getStatus();
	}
	
	public Originator(String status) {
		super();
		this.status = status;
	}
	
	public void printStatus() {
		System.out.println(status);
	}
	
	public void updateStatus(String status) {
		this.status = status;
	}
}

File: cplusplus/design_pattern/MomentoProj/Originator.cpp
#include "stdafx.h"
#include "Originator.h"


Originator::Originator()
{
}


Originator::~Originator()
{
}


Originator::Originator(std::string status) : status(status)
{
}

void Originator::updateStatus(std::string status)
{
    this->status = status;
}


void Originator::printStatus()
{
    std::cout << status.c_str() << std::endl;
}


std::shared_ptr<Momento> Originator::createMomento()
{
    return std::make_shared<Momento>(status);
}


void Originator::restoreStatus(std::shared_ptr<Momento> momento)
{
    status = momento->getStatus();
}
